lib/kitchen/driver/vagrant.rb

```diff
@@ -310,12 +310,20 @@ def run(cmd, options = {})
       # * `:log_subject` defaults to a String representation of the Driver's
       #   class name
       #
+      # Since vagrant does not support being run through bundler, we escape
+      # any bundler environment should we detect one.  Otherwise, subcommands
+      # will inherit our bundled environment.
+      # @see https://github.com/test-kitchen/kitchen-vagrant/issues/190
       # @see Kitchen::ShellOut#run_command
       def run_command(cmd, options = {})
         merged = {
           :use_sudo => config[:use_sudo], :log_subject => name
         }.merge(options)
-        super(cmd, merged)
+        if Object.const_defined?("Bundler")
+          Bundler.with_clean_env { super(cmd, merged) }
```

Comment:
I terribly misunderstood how bundler (and bundle exec works).  Bundler.with_clean_env is utterly useless in the subprocess because by the time it has exec-ed the underlying script, the "pristine" environment is long gone.  PATH has been mangled and GEM_* has been reset.  And ORIGINAL_GEM_PATH is anything but original.

You're right...  this is very very sad :(